File: Diseño de Interfaces/ListView con CheckBox/Versionesadapter.java
package com.example.deportes;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;

public class Versionesadapter extends BaseAdapter
{

    private LayoutInflater mInflater;
    private ArrayList <Androidversion> versiones;

    public Versionesadapter(Context context, ArrayList< Androidversion> vers) {
        super();
        this.mInflater = LayoutInflater.from(context);
        this.versiones = vers;
    }

    public int getCount() {
        return versiones.size();
    }
    public Androidversion getItem(int position)
    {
        return versiones.get(position);
    }
    public long getItemId(int position)
    {
        return position;
    }

    public View getView(int position, View convertView, ViewGroup parent)
    {
        ViewHolder holder = null;
        if (convertView == null)
        {
            convertView = mInflater.inflate(R.layout.lista_deportes, null);
            holder = new ViewHolder();
            holder.hNombre = (TextView) convertView.findViewById(R.id.idNombre);
            holder.hImage = (ImageView) convertView.findViewById(R.id.idImage);
            convertView.setTag(holder);
        }
        else
        {
            holder = (ViewHolder) convertView.getTag();
        }

        Androidversion version = getItem(position);
        holder.hNombre.setText(version.getNombre());
        holder.hImage.setImageResource(version.getLogo());
        return convertView;
    }

    class ViewHolder
    {
        TextView hNombre;
        ImageView hImage;
    }
}

File: Diseño de Interfaces/menus/Menu Lateral, Menu con pestañas, fragment deslizante y Menu Contextual/Kotlin/VersionesAdapter.kt
package com.example.segundaevaluacion

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView

class VersionesAdapter(context: Context, private val versiones: ArrayList<AndroidVersion>) : ArrayAdapter <AndroidVersion>(context, R.layout.pagina2, versiones)
{
    private val mInflater: LayoutInflater

    init {
        this.mInflater = LayoutInflater.from(context)
    }

    override fun getCount(): Int {
        return versiones.size
    }

    override fun getItem(position: Int): AndroidVersion? {
        return versiones[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        var holder: ViewHolder? = null

        if (convertView == null) {
            convertView = mInflater.inflate(R.layout.list_view, null)
            holder = ViewHolder()
            holder.hNombre = convertView.findViewById(R.id.textNombre) as TextView
            holder.hNumero = convertView.findViewById(R.id.textNum) as TextView
            holder.hImage = convertView.findViewById(R.id.imagefoto) as ImageView
            convertView.tag = holder
        }
        else {
            holder = convertView!!.tag as ViewHolder?
        }
        val usuarios = getItem(position)
        holder!!.hNombre!!.text = usuarios!!.nombre
        holder.hNumero!!.text = usuarios.numero
        holder.hImage!!.setImageResource(usuarios.logo)
        return convertView!!
    }

    internal inner class ViewHolder
    {
        var hNombre: TextView? = null
        var hNumero: TextView? = null
        var hImage: ImageView? = null
    }
}